libr/anal/p/anal_xtensa.c

```diff
@@ -634,6 +634,12 @@ static inline void sign_extend(st32 *value, ut8 bit) {
 	}
 }
 
+static inline void sign_extend2(st32 *value, ut8 bit1, ut8 bit2, ut8 shift) {
+	ut8 bit = ((*value >> bit1) & 1) & ((*value >> bit2) & 1);
+	if(bit)
+		*value |= 0xFFFFFFFF << (31 - shift);
```

Comment:
use UT32_MAX insteadd of 0xffff

Reply:
In such case sign_extend basing which I did my sign_extend2 should use UT32_MAX too (in this or by separate PR)